builtin/rebase--interactive.c

```diff
@@ -185,6 +185,8 @@ int cmd_rebase__interactive(int argc, const char **argv, const char *prefix)
 		{ OPTION_STRING, 'S', "gpg-sign", &opts.gpg_sign, N_("key-id"),
```

Comment:
Ah! I was basing my work on [`v2.21.0`](https://github.com/git/git/tree/v2.21.0). Ok, I'll rebase it to master and then v2.22.0 when it comes out.

Reply:
You should always base your work on master (and keep a look out at other branches that might conflict with your work).  Sometimes it's better to base your work on a branch that has not been merged to `master` yet to avoid conflicts.  Note that in this case your work can't be merged to master before the other branch is merged.

For bug fixes you can also sometimes base your work on `maint`, from which maintenance releases are created.  But that's only necessary for bug fixes really.

Also note that once you sent your patch series to the mailing list for the first time, you should not rebase it anymore, unless there is a very good reason for it, such as major conflicts.  That should be noted in the cover letter in this case.  Junio usually prefers solving smaller conflicts himself, and keeping the same base for a branch.